src/UCTNode.cpp

```diff
@@ -92,8 +92,36 @@ bool UCTNode::create_children(Network & network,
             legal_sum += raw_netlist.policy[i];
         }
     }
-    nodelist.emplace_back(raw_netlist.policy_pass, FastBoard::PASS);
-    legal_sum += raw_netlist.policy_pass;
+
+    // Add pass move. BUT don't do this if the following conditions
+    // all obtain (see issue #2273, "Double-passing pathology"):
+    //   - The NN's evaluation of the current node is very good
+    //     for the player whose move it is. (Say, 0.75 or better.)
+    //     (So we don't want to end the game unless we win.)
+    //   - Ending the game now would actually lose the game for
+    //     the player whose move it is.
+    //     (So we don't want to end the game.)
+    //   - We do have at least five other legal moves.
+    //     (So it's not likely that all our available moves
+    //     are actually disastrous.)
+    //   - The "dumbpass" option is not turned on.
+    //     (Because, as per GCP's comment at
+    //     https://github.com/leela-zero/leela-zero/issues/2273#issuecomment-472398802 ,
+    //     enabling this heuristic is un-Zero-like and enabling
+    //     dumbpass is meant to suppress any such things that
+    //     affect passing.)
+    // The magic numbers 0.75 and 5 are somewhat arbitrary and it seems
+    // unlikely that their values make much difference.
+    // This check prevents some serious evaluation errors but has a cost:
+    // we make extra calls to final_score() at some nodes. This is much
+    // cheaper than a single NN evaluation, and the cost should be negligible.
```

Comment:
This is a long comment and it doesn't add much, as it's literally reading off the conditions of the if that follows. I think it's better to put this in the commit message or just refer to the issue here.

Comments are very good if they're explaining something that is not obvious from the code, but here it's just repeating what the code does. I'd rather make the code clearer.

Reply:
There's some actual explanation there as well, but I agree we can get by with substantially less.